common/djangoapps/student/models.py

```diff
@@ -200,12 +200,18 @@ def anonymous_id_for_user(user, course_id, save=True):
         monitoring.increment('temp_anon_uid_v2.fetched_existing')
     else:
         # include the secret key as a salt, and to make the ids unique across different LMS installs.
-        hasher = hashlib.md5()
+        hasher = hashlib.shake_128()
+        # This is one of several uses of SECRET_KEY.
+        #
+        # Impact of exposure: If a person has the SECRET_KEY and a user's `id`
+        # they can correlate the users anonymous user IDs across any courses they have participated in.
+        #
+        # Rotation process: Can be rotated at will.
```

Comment:
```suggestion
        # Rotation process: Can be rotated at will. There is a small chance (on the order of 1%) that
        # any given new user will be assigned multiple anonymous user IDs during the period in which
        # servers are configured with a mix of old and new keys.
```
I'm realizing now that we had documented this a little optimistically. We're not out of the woods yet in terms of rotation being safe.